### PR TITLE
Issue 137 tag completion

### PR DESCRIPTION
**Pull request by Bruce-Connor**
Implements a new command, bound to C-c C-q in compose-mode, which inserts tags with completion.

### CODE REVIEW

File: sx-search.el [hunk 1]
```diff
@@ -84,15 +82,12 @@ prefix argument, the user is asked for everything."
      (when (string= query "")
        (setq query nil))
      (when current-prefix-arg
-       (setq tags (sx--multiple-read
-                   (format "Tags (%s)"
-                     (if query "optional" "mandatory"))
-                   'sx-search--tag-history))
+       (setq tags (sx-tag-multiple-read
+                   site (format "Tags%s" (if query " (optional)" ""))))
```
Comment (vermiculus) [inline sx-search.el]: You can use `concat` here instead of `format`

File: sx-tag.el [hunk 2]
```diff
@@ -82,6 +99,40 @@ Return the list of invalid tags in TAGS."
...
     (cl-remove-if (lambda (x) (member x result)) tags)))
 
+

+;;; Prompt the user for tags.
+(defvar sx-tag-history nil
+  "Tags history for interactive prompts.")
+
+;;; @TODO: Make it so that hitting BACKSPACE with an empty input
+;;; deletes a previously submitted tag.
```
Comment (vermiculus) [inline sx-tag.el]: :+1:

File: sx.el [hunk 3]
```diff
@@ -265,6 +247,18 @@ Anything before the (sub)domain is removed."
     (rx string-start (or (and (0+ word) (optional ":") "//")))
     "" url)))
 
+(defmacro sx--define-conditional-key (keymap key def &rest body)
+  "In KEYMAP, define key sequence KEY as DEF conditionally.
+This is like `define-key', except the definition \"disapears\"
+whenever BODY evaluates to nil."
```
Comment (vermiculus) [inline sx.el]: `disappears`

File: sx-request.el [hunk 4]
```diff
@@ -213,6 +213,35 @@ Currently returns nil."
...
+    (if (not response-buffer)
+        (error "Something went wrong in `url-retrieve-synchronously'")
+      (with-current-buffer response-buffer
+        (progn
+          (goto-char (point-min))
+          (if (not (search-forward "\n\n" nil t))
+              (error "Headers missing; response corrupt")
+            (when (looking-at-p "Not Found") (error "Page not found."))
+            (prog1 (read (current-buffer))
+              (kill-buffer (current-buffer)))))))))
```
Comment (vermiculus) [inline sx-request.el]: Should we also be killing the buffer we create with standard `sx-request-make`? I didn't think the buffer would be persistent, but I don't know why I thought that.

Comment (Bruce-Connor) [inline sx-request.el]: Yes we should. I was going to post that as an issue but forgot. 
My session had over a hundred left over buffers.

File: sx-search.el (hunk 1 above)
Comment (Bruce-Connor) [inline sx-search.el]: Will do

File: sx-request.el (hunk 4 above)
Comment (vermiculus) [inline sx-request.el]: :exclamation:

Comment (vermiculus) [inline sx-request.el]: #207

**Comment by Bruce-Connor** [comment]
Changes implemented. Feel free to merge if you agree.

**Pull request merged**